### PR TITLE
added new command line options related to preview output

### PR DESCRIPTION
**Pull request by doughadfield**
modified raspimjpeg.py file:
Added -p (--preview) to enable HDMI port video output
Added --s_height and --s_width to separately specify resolution of video
stream
if not set, these default to the same as the video device settings (-w,
-h)
New streams_iter() function to speed up frame processing in network
stream
Requires addition to "Video Device" web form to add "preview" switch (boolean) to set -p flag, and resolution drop-down in "Video Streaming" form to set s_width and s_height

To do: work out a scheme for separately setting frame rate for network video stream

### CODE REVIEW

**Comment by doughadfield** [comment]
added arguments to raspimjpeg.py process to enable HDMI preview output. By default behaviour is unchanged from previous version. 
incremented version count in file to 0.6 (from 0.5)

File: extras/raspimjpeg.py [hunk 1]
```diff
@@ -228,13 +222,23 @@ def init_camera():
...
+    while running:
+	yield sys.stdout
+	sys.stdout.flush()
+
+
 def run():
     logging.debug('starting capture')
-    camera.capture_sequence(streams_iter(), format='jpeg',
-            use_video_port=not options.stills, thumbnail=None, quality=options.quality)
+    camera.capture_sequence(streams_iter(), format='jpeg', resize=(options.s_width, options.s_height), use_video_port=not options.stills, thumbnail=None, quality=options.quality)
```
Comment (doughadfield) [inline extras/raspimjpeg.py:241]: Need to split this into two lines, to reduce page width. Sorry, missed this before commit

**Review by doughadfield — COMMENTED** [review]
will tidy up appearance when next changes needed

**Comment by ccrisan** [comment]
1. Why have you changed "capture" to "camera" in the context of width and height arguments? It still is the capturing resolution even after your changes. And when you say "camera width/height" you usually think of the maximum supported camera resolution.
2. I really don't like the names of "s_width/s_height" arguments. Let's go with `-x`, `--streaming-width` and `-y`, `--streaming-height`. Command line arguments rarely make use of underscore.
3. The preview functionality is perfect.
4. We should see if we can actually go beyond 1920x1080 and in what circumstances, as people have asked for higher resolutions knowing that the camera actually does support up to 2k something. It may only work in stills mode.

**Comment by doughadfield** [comment]
Hi Calin, I agree with all your comments – you know the product better than anyone and any naming scheme should be consistent with what you’ve already used. I changed the name of “capture” to “camera” as to me “capture” seemed to imply the streaming video (which would be the version captured to disk if that was possible in “fast network camera” mode). Whereas the native resolution of the camera, which is output to the HDMI port, is just that  - the native camera resolution. The “camera” resolution setting is ONLY seen in the HDMI output, as it’s the streaming resolution that would be used in any capture or remote display application.

If you’re using the word “capture” to mean image capture by the camera module, then your name makes perfect sense, and we should perhaps call the two resolutions “capture” and “streaming”.

If you want to change the names, of course it’s up to you. I’d be happy to do it for you if you don’t have time.

I hope I’ve been able to help a bit – I’d be very happy to assist in any other way with this most excellent project! :)

Please let me know if there’s anything more I can do…

```
            Doug
```

**Comment by doughadfield** [comment]
Calin, I need to apologise to you for committing my change before giving you the opportunity to review the changes. I'm new to Github contribution, so I went too far in the workflow by mistake. If you allow me to make any future additions to the product, I'll be sure to wait for your review before committing any changes.
Doug

**Comment by ccrisan** [comment]
Sure, no worries. This is why I granted you commit access to the repo and this is why we have this comments system in place :)

**Comment by doughadfield** [comment]
Calin, I'll make the argument name changes you suggest, as soon as I'm on a stable internet connection, so the arguments are established before too many people start using them ;-)

**Comment by ccrisan** [comment]
I took care of the renaming: we now have `--preview-width` and `--preview-height` or `-x` and `-y`, respectively.

**Comment by ccrisan** [comment]
For the record, https://github.com/ccrisan/motioneyeos/commit/ffbc49a5d6d3c57814959217125d7b318fed8ee5 adds support for HDMI preview and many more.

**Pull request merged**